File: framework/src/main/java/com/thornbirds/frameworkext/component/page/PageParams.java
package com.thornbirds.frameworkext.component.page;

import com.thornbirds.component.Params;
import com.thornbirds.frameworkext.component.route.IPageParams;

/**
 * @author YangLi yanglijd@gmail.com
 */
public class PageParams extends Params implements IPageParams {
    @Override
    public PageParams putItem(Object object) {
        return (PageParams) super.putItem(object);
    }

    public static PageParams from(Object... values) {
        if (values != null && values.length > 0) {
            final PageParams params = new PageParams();
            for (Object elem : values) {
                params.putItem(elem);
            }
            return params;
        }
        return null;
    }
}

File: framework/src/main/java/com/thornbirds/frameworkext/component/page/PageParams.kt
package com.thornbirds.frameworkext.component.page

import com.thornbirds.component.Params
import com.thornbirds.frameworkext.component.route.IPageParams

/**
 * @author YangLi yanglijd@gmail.com
 */
class PageParams : Params(), IPageParams {

    override fun putItem(item: Any): PageParams {
        return super.putItem(item) as PageParams
    }

    companion object {
        fun from(vararg values: Any): PageParams? {
            return if (values.isNotEmpty()) {
                PageParams().also {
                    values.forEach { elem ->
                        it.putItem(elem)
                    }
                }
            } else null
        }
    }
}